Makefile.include

```diff
@@ -242,6 +242,16 @@ ifneq (, $(filter $(GLOBAL_GOALS), $(MAKECMDGOALS)))
   include $(RIOTMAKE)/buildtests.inc.mk
 else
 
+
+.PHONY: CREATECACHEDIR
+
+CREATECACHEDIR:
+	$(Q)mkdir -p "$(BUILD_DIR)"
+	$(Q)echo "Signature: 8a477f597d28d172789f06886806bc55" > "$(BUILD_DIR)/CACHEDIR.TAG"
+	$(Q)echo "# This folder contains RIOT's build cache" >> "$(BUILD_DIR)/CACHEDIR.TAG"
+
+-include CREATECACHEDIR
```

Comment:
This is a novel/interesting use of the include directive. You may consider adding a comment explaining the intent.

Reply:
I saw what you said earlier about the `order only prerequisites` but I don't think it's the best way to handle this as it means that every new target that will be added in the future will need to ensure it includes this specific target as a prequisite.

On the opposite side what I've done using `-include CREATECACHEDIR` will call the target `CREATECACHEDIR` regardless of the original target called by the user which is the use case we've identified so far. Even if it's probably not the cleanest way to do that I think that the maintainability is a really strong argument to its usage.

Reply:
its probably a good idea to create the directory when you need it -> depend on its or the tag file existence

Reply:
Ahh i see it is the `mkdir -p` in the pkg.mk which create the directory without the tag 

probably the right solution would be to for `$(PKG_DOWNLOADED):` to depend on the CACHEDIR.TAG file  after checking if the state-files share a part of the path with the CACHEDIR.TAG

Reply:
I think that we also previously discussed the fact that there might also be some other cache directories. Like other directories with a CACHEDIR.TAG that are not build, which motivated my choice of creating a new target for this specific job. It means that whenever a new directory with a CACHEDIR.TAG will be needed it will only need to be added as a prerequisite in the target I created.